cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,356 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/kueue/client-go/clientset/versioned/scheme"
+	"sigs.k8s.io/kueue/cmd/kueuectl/app/util"
+	kueuejob "sigs.k8s.io/kueue/pkg/controller/jobs/job"
+	kueuejobset "sigs.k8s.io/kueue/pkg/controller/jobs/jobset"
+	kueuemxjob "sigs.k8s.io/kueue/pkg/controller/jobs/kubeflow/jobs/mxjob"
+	kueuepaddlejob "sigs.k8s.io/kueue/pkg/controller/jobs/kubeflow/jobs/paddlejob"
+	kueuepytorchjob "sigs.k8s.io/kueue/pkg/controller/jobs/kubeflow/jobs/pytorchjob"
+	kueuetfjob "sigs.k8s.io/kueue/pkg/controller/jobs/kubeflow/jobs/tfjob"
+	kueuexgboostjob "sigs.k8s.io/kueue/pkg/controller/jobs/kubeflow/jobs/xgboostjob"
+	kueuempijob "sigs.k8s.io/kueue/pkg/controller/jobs/mpijob"
+	"sigs.k8s.io/kueue/pkg/controller/jobs/pod"
+	kueueraycluster "sigs.k8s.io/kueue/pkg/controller/jobs/raycluster"
+	kueuerayjob "sigs.k8s.io/kueue/pkg/controller/jobs/rayjob"
+)
+
+const (
+	podLong = `Lists all pods that matches the given criteria: Should be part of the specified Job kind,
+	belonging to the specified namespace, matching
+	the label selector or the field selector.`
+	podExample = `  # List Pods
+  kueuectl list pods --for job/job-name`
+)
+
+var jobControllersWithPodLabelSelector = []jobControllerWithPodLabelSelector{
+	&kueuejob.Job{},
+	&kueuejobset.JobSet{},
+	&kueuemxjob.JobControl{},
+	&kueuepaddlejob.JobControl{},
+	&kueuetfjob.JobControl{},
+	&kueuepytorchjob.JobControl{},
+	&kueuexgboostjob.JobControl{},
+	&kueuempijob.MPIJob{},
+	&pod.Pod{},
+	&kueueraycluster.RayCluster{},
+	&kueuerayjob.RayJob{},
+}
+
+type PodOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	AllNamespaces          bool
+	Namespace              string
+	LabelSelector          string
+	FieldSelector          string
+	UserSpecifiedForObject string
+	ForName                string
+	ForGVK                 schema.GroupVersionKind
+	ForObject              *unstructured.Unstructured
+	PodLabelSelector       string
+
+	Clientset k8s.Interface
+
+	genericiooptions.IOStreams
+}
+
+type jobControllerWithPodLabelSelector interface {
+	Object() client.Object
+	GVK() schema.GroupVersionKind
+	PodLabelSelector() string
+}
+
+func NewPodOptions(streams genericiooptions.IOStreams) *PodOptions {
+	return &PodOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewPodCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewPodOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "pods --for TYPE[.API-GROUP]/NAME",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"po"},
+		Short:                 "List Pods belong to a Job Kind",
+		Long:                  podLong,
+		Example:               podExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter))
+			if o.ForObject != nil {
+				cobra.CheckErr(o.Run(cmd.Context(), clientGetter))
+			}
+		},
```

Comment:
```suggestion
RunE: func(cmd *cobra.Command, args []string) error {
			cmd.SilenceUsage = true
			err := o.Complete(clientGetter)
			if err != nil {
				return err
			}
			if o.ForObject == nil {
				return nil
			}
			return o.Run(clientGetter)
		},
```

We already fixed another commands #2578.